PVTPackage/source/MultiphaseSystem/PhaseSplitModel/DeadOilFlash.cpp

```diff
@@ -34,9 +36,34 @@ DeadOilFlash::DeadOilFlash( std::vector< std::vector< double > > const & PVDO,
   m_gasPhaseModel( pvt::PHASE_TYPE::GAS, PVDG, gasSurfaceMassDensity, gasSurfaceMolecularWeight ),
   m_waterPhaseModel( PVTW, waterSurfaceMassDensity, waterSurfaceMolecularWeight )
 {
+}
 
+DeadOilFlash::DeadOilFlash( std::vector< std::vector< double > > const & PVDO,
+                            double oilSurfaceMassDensity,
+                            double oilSurfaceMolecularWeight,
+                            std::vector< std::vector< double > > const & PVDG,
+                            double gasSurfaceMassDensity,
+                            double gasSurfaceMolecularWeight )
+  :
+  m_oilPhaseModel( pvt::PHASE_TYPE::OIL, PVDO, oilSurfaceMassDensity, oilSurfaceMolecularWeight ),
+  m_gasPhaseModel( pvt::PHASE_TYPE::GAS, PVDG, gasSurfaceMassDensity, gasSurfaceMolecularWeight ),
+  m_waterPhaseModel()  
+{
 }
 
+DeadOilFlash::DeadOilFlash( std::vector< std::vector< double > > const & PVDO,
+                            double oilSurfaceMassDensity,
+                            double oilSurfaceMolecularWeight,
+                            std::vector< double > const & PVTW,
+                            double waterSurfaceMassDensity,
+                            double waterSurfaceMolecularWeight )
+  :
+  m_oilPhaseModel( pvt::PHASE_TYPE::OIL, PVDO, oilSurfaceMassDensity, oilSurfaceMolecularWeight ),
+  m_gasPhaseModel( pvt::PHASE_TYPE::GAS ),  
```

Comment:
...neither this `m_gasPhaseModel`. Am I correct?

Reply:
Yes you are correct in an oil-water simulation, we would not use `m_gasPhaseModel`.

Reply:
Then what about transforming `m_waterPhaseModel` and `m_gasPhaseModel` into pointers and allocate them only if needed.
That would prevent the two dummy constructors. and since we are encapsulated, `getWaterModel` and `getGasModel` will hide this.
Any call to these models would crash, but any call would result from a wrong simulation so...